addons/l10n_ar_withholding/models/account_payment.py

```diff
@@ -1,9 +1,35 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
-from odoo import models, fields
+from odoo import api, models, fields, _
+from odoo.exceptions import ValidationError
 
 
 class AccountPayment(models.Model):
 
     _inherit = 'account.payment'
 
     l10n_ar_withholding_ids = fields.One2many(related='move_id.l10n_ar_withholding_ids')
+
+    def _synchronize_to_moves(self, changed_fields):
+        ''' If we change a payment with withholdings, delete all withholding lines as the synchronization mechanism is not
+        implemented yet
+        '''
+        if not any(field_name in changed_fields for field_name in self._get_trigger_fields_to_synchronize()):
+            return
+        for pay in self:
+            pay.line_ids.filtered(
+                lambda
+                x: x.account_id == pay.company_id.l10n_ar_tax_base_account_id or
+                x.tax_line_id.l10n_ar_withholding_payment_type
+            ).unlink()
+        res = super()._synchronize_to_moves(changed_fields)
+        return res
+
+    @api.constrains('state', 'move_id')
+    def _check_move_id(self):
+        for payment in self:
+            if (
+                not payment.move_id and
+                payment.payment_method_code in ('own_checks', 'new_third_party_checks', 'in_third_party_checks', 'out_third_party_checks') and
+                not payment.outstanding_account_id
+            ):
+                raise ValidationError(_("A payment with any Third Party Check or Own Check payment methods needs an outstanding account"))
```

Comment:
This constraint should be defined in the check module, not in the witholding module.